edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/publisher/LogAnalyticsUpload.cs

```diff
@@ -39,5 +41,21 @@ public async Task<bool> PublishAsync(IEnumerable<Metric> metrics, CancellationTo
                 return false;
             }
         }
+
+        class MetricWithStringTags
```

Comment:
I believe that our coding standards specify 1 class per file. Split this out?

Reply:
Also, why not just call this MetricWithTags? String is leaking implementation details.

Reply:
This class is a result of changing the metric class Tags from a string to a Dictionary. I think it should remain a private class nested in LogAnalyticsUpload, since it's a very slight transformation of the data that only makes sense just prior to upload. I will rename it to UploadMetric.

Reply:
If that's enforced by our coding pattern I can switch it. I'd like to keep it's scope small though. It should really only exist for LogAnalyticsUpload

Reply:
I could also change it to 
```
var metricsToUpload = metrics.Select(m => new { m.TimeGeneratedUtc, m.Name, m.Value, Tags = JsonConvert.SerializeObject(m.Tags) });
```
and not have a declared class